alacritty/src/macos/locale.rs

```diff
@@ -61,38 +58,15 @@ fn system_locale() -> String {
         // https://developer.apple.com/documentation/foundation/nslocale/1416263-localeidentifier?language=objc
         // https://developer.apple.com/documentation/foundation/nslocale/1643060-countrycode?language=objc
         // https://developer.apple.com/documentation/foundation/nslocale/1643026-languagecode?language=objc
-        let is_language_code_supported: bool =
-            msg_send![locale, respondsToSelector: sel!(languageCode)];
-        let is_country_code_supported: bool =
-            msg_send![locale, respondsToSelector: sel!(countryCode)];
-        let locale_id = if is_language_code_supported && is_country_code_supported {
-            let language_code: *const Object = msg_send![locale, languageCode];
-            let language_code_str = nsstring_as_str(language_code).to_owned();
-            let _: () = msg_send![language_code, release];
-
-            let country_code: *const Object = msg_send![locale, countryCode];
-            let country_code_str = nsstring_as_str(country_code).to_owned();
-            let _: () = msg_send![country_code, release];
-
-            format!("{}_{}.UTF-8", &language_code_str, &country_code_str)
+        let is_language_code_supported: bool = locale.respondsToSelector(sel!(languageCode));
+        let is_country_code_supported: bool = locale.respondsToSelector(sel!(countryCode));
+        if is_language_code_supported && is_country_code_supported {
+            let language_code = locale.languageCode();
+            #[allow(deprecated)]
+            let country_code = locale.countryCode().unwrap();
```

Comment:
This unwrap does not sound right to me. This is deprecated, is there plan to remove it potentially which could cause this to fail? We shouldn't crash here under any circumstances.

Reply:
The nullability has nothing to do with it being deprecated.

This was nullable before, you just weren't handling it (which was potentially UB).

Reply:
Seems easy enough to handle, so I don't see why we wouldn't. Unwrap's not the right choice.

Reply:
How do you want me to handle it?

Reply:
Just falling back to the `else` branch if anything goes wrong is fine, should be the simplest solution. I'd just early return on success probably.

Reply:
But... The comment above says:
> `localeIdentifier` returns extra metadata with the locale (including currency and collator) on newer versions of macOS. This is not a valid locale, so we use `languageCode` and `countryCode`, if they're available (macOS 10.12+):

So we can't fall back to `localeIdentifier` on newer versions of macOS?

Reply:
That's a good point. Falling back to en_US.UTF-8 is also fine or whatever you think is a reasonable default. I think the assumption here is this is almost never hit anyway, so we just don't want to crash really. Bugs would be easier to work around for users than crashes.

Reply:
I've done that now, then